packages/material-ui/src/Portal/Portal.js

```diff
@@ -28,13 +28,21 @@ const Portal = React.forwardRef(function Portal(props, ref) {
     }
   }, [container, disablePortal]);
 
-  React.useImperativeHandle(ref, () => mountNode || childRef.current, [mountNode]);
+  useEnhancedEffect(() => {
+    if (mountNode || disablePortal) {
+      setRef(ref, mountNode || childRef.current);
```

Comment:
That's a leaky usage of setRef again. Previous usage was fine. What broke without this change?

Reply:
The test diff showcases the change. The ref was first called with null. The Modal logics need the first call to contain a valid value. The ref is cleanup in the teardown. Why would it leak?

Reply:
childRef.current is not part of the deps array.